### PR TITLE
Protected tag is no internal server error

### LINKED ISSUE #30959: Weird HTTP code when invalid tag name is provided
### Description

When I try to create tag with name that isn't allowed due to tag protection, I'm receiving 500 status code. I think it should be something like 401 (Unauthorized) or 400 (Bad request). 

I've set up a step in my Gitea Action to create new release in my repository.
Honestly, when I saw "Internal Server Error" in my step logs I thought that something is wrong with my server.
But it was only poorly written regular expression.

### Gitea Version

1.21.11

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

https://gist.github.com/sadorowo/2f212589fe30816f7d1d882d06fbdbda

### Screenshots

Not related to web interface, but I've made request from Developer Tools

image

### Git Version

2.40.1, Wire Protocol Version 2 Enabled

### Operating System

Ubuntu 22.04.4 LTS x86_64

### How are you running Gitea?

I'm using Docker Compose version of Gitea.

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #30959

Adds an API test for protected tags.
Fix existing tag in combination with fixtures.

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: routers/api/v1/repo/release.go [hunk 1]
```diff
@@ -246,6 +246,8 @@ func CreateRelease(ctx *context.APIContext) {
 		if err := release_service.CreateRelease(ctx.Repo.GitRepo, rel, nil, ""); err != nil {
 			if repo_model.IsErrReleaseAlreadyExist(err) {
 				ctx.Error(http.StatusConflict, "ReleaseAlreadyExist", err)
+			} else if models.IsErrProtectedTagName(err) {
+				ctx.Error(http.StatusMethodNotAllowed, "ProtectedTagName", err)
```
Comment (silverwind) [inline routers/api/v1/repo/release.go]: Woudn't https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403 be more suitable?

**Comment by silverwind** [comment]
I think you need to alter swagger comment docs to indicate the possibility for any new status codes.

File: routers/api/v1/repo/release.go (hunk 1 above)
Comment (silverwind) [reply routers/api/v1/repo/release.go]: If we follow [GH API](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release), it should be 422 Conflict:

image

Comment (KN4CK3R) [reply routers/api/v1/repo/release.go]: Yes, another status code is better. Just used the one all other places use.

Comment (silverwind) [reply routers/api/v1/repo/release.go]: Not sure if it's wise to diverge from GH API, to keep things like actions compatible. I would prefer if we change to 422.

**Review by silverwind — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**